src/iexamine.cpp

```diff
@@ -630,22 +630,65 @@ void iexamine::cardreader(player *p, map *m, int examx, int examy)
 void iexamine::rubble(player *p, map *m, int examx, int examy)
 {
     bool has_digging_tool = p->has_items_with_quality( "DIG", 2, 1 );
-  // Perhaps check for vehicle covering the rubble and bail out if so (string freeze ATM)?
+    // Perhaps check for vehicle covering the rubble and bail out if so (string freeze ATM)?
     if (!has_digging_tool) {
         add_msg(m_info, _("If only you had a shovel..."));
         return;
     }
+
+    // Ask if there's something possibly more interesting than this rubble here
     std::string xname = m->furnname(examx, examy);
-    if (query_yn(_("Clear up that %s?"), xname.c_str())) {
-        // "Remove"
-        p->moves -= 200;
-        m->furn_set(examx, examy, f_null);
+    if( ( m->veh_at( examx, examy ) != nullptr ||
+          m->tr_at( examx, examy ) != tr_null ||
+          g->critter_at( examx, examy ) != nullptr ) &&
+          !query_yn(_("Clear up that %s?"), m->tername(examx, examy).c_str() ) ) {
+        none(p, m, examx, examy);
+        return;
+    }
+
+    // "Remove"
+    p->moves -= 200;
+    m->furn_set(examx, examy, f_null);
+
+    // "Remind"
+    add_msg(_("You clear up that %s."), xname.c_str());
+}
+
+void iexamine::crate(player *p, map *m, int examx, int examy)
+{
+    // Check for a crowbar in the inventory
+    const auto has_prying = []( const item it ) {
+        const auto fun = it.type->get_use( "CROWBAR" );
```

Comment:
One line: `return ...->has_use(...)`

Btw.  there is nothing wrong with casting the const away from the item returned by `Character::items_with`, just use a `const_cast`. Doing this is only problematic if the object itself was defined as const like this:
```C++
const int a = 10;
int &b = const_cast<int&>(a);
b = 20; // undefined behavior

int t = 100;
const int &v = t;
int &x = const_cast<int&>(v);
x = 200; // fine, because x==v==t and t is not const
```